docs/tangle.md

```diff
@@ -0,0 +1,276 @@
+
+    
+## Solidification
+A message is said to be **solid** on a node when all its parents are known to the node and also marked as solid.
+
+If a node is missing a referenced message it is stored in the **solidification buffer** and not yet processed. A node can ask its neighbors for the missing message by sending a **solidification request** containing the message hash. This process can be recursively repeated until all of a message's past cone until the genesis (or snapshot) become solid which is known as **solidification**. In that way the Tangle enables even nodes joining the network at a point later in time to retrieve all of a message's history.
+
+### Naive approach
+Approach:
+1. Send solidification request for message immediately to all neighbors
+2. If not received after `solidificationRetryInterval` seconds, send solidification request again. Repeat until received.
+
+This approach simply requests missing messages recursivley until all of them become solid. While easy to implement it has some drawbacks. Each  parent that needs to be requested adds another RTT and message complexity (solidification request * neighbors). 
+
+Optimizations:
+- do not send solidification request immediately: messages can arrive out of order but should generally arrive within a small time window since other nodes only gossip messages on solidification
+- instead of asking for messages one by one a node could ask for messages of a certain timeframe (e.g. its local snapshot time until now)
+- send solidification request only to a subset of neighbors
+
+### Possible attacks
+A malicious node can send unsolidifiable messages to a node. A simple protection is to only repeat solidification requests `solidificationMaxRepeat` times before the message gets deleted from the node's solidification buffer. 
+
+
+## Orphanage
+Messages that are considered to be invalid/not eligible are discarded and are consequently ignored during tip selection. This process of leaving undesired messages (or more in general, whenever a message is not being approved) behind is called **orphaning messages**. It is important that nodes share the same perception on which messages should be orphaned. 
+
+Orphaned messages can be safely deleted during snapshotting and are not visible to nodes that later join the network because they are not reachable when requesting the missing messages from the tips during solidification.
+
+## Finality
+Users need to know when their information has been successfully added to the Tangle. In other words, they need to know when their information will not be orphaned. However, finality is inherently probabilistic. For instance, consider the following scenario. An attacker can trivially maintain a chain of messages that do not approve any other message. At any given point in time, it is possible that all messages will be orphaned except this chain. This is incredibly unlikely, but yet still possible.
```

Comment:
"Users need to know when their information has been successfully added to the Tangle. In other words, they need to know when their information will not be orphaned"

Messages added to the Tangle can still be orphaned, don't they?